viewer/src/glsl/base/determineColor.glsl

```diff
@@ -0,0 +1,20 @@
+// TODO specify width and height using uniforms
+const float dataTextureWidth = 2048.0;
+const float dataTextureHeight = 2048.0;
+
+vec3 determineColor(vec3 originalColor, sampler2D colorDataTexture, float treeIndex) {
+    float u = mod(treeIndex, dataTextureWidth);
+    float v = floor(treeIndex / dataTextureWidth);
+    float uCoord = (u + 0.5) / dataTextureWidth;
+    float vCoord = (v + 0.5) / dataTextureHeight; // invert Y axis
+    vec2 treeIndexUv = vec2(uCoord, vCoord);
+    vec4 overrideColor = texture2D(colorDataTexture, treeIndexUv);
+
+    if (overrideColor.a == 0.0) {
+        return originalColor;
+    }
+
+    return overrideColor.rgb;
```

Comment:
`return ceil(overrideColor.a) * overrideColor.rgb + floor(overrideColor.a)*originalColor` for branch-free bliss.

Reply:
I do not think this works:

- if `overrideColor.a` is `0.0`, then `floor(0.0)` and `ceil(0.0)` will both return `0.0`? Then the color is `rgb(0, 0, 0)`.
- if `overrideColor.a` is `1.0`, then we get `1.0` on both, which is a mix of the two colors.

This should work, however:

```
return overrideColor.a * overrideColor.rgb + (1.0 - overrideColor.a) * originalColor;
```